src/GoTrueClient.ts

```diff
@@ -357,11 +357,13 @@ export default class GoTrueClient {
    */
   async signInWithOAuth(credentials: SignInWithOAuthCredentials): Promise<OAuthResponse> {
     await this._removeSession()
+
     return this._handleProviderSignIn(credentials.provider, {
       redirectTo: credentials.options?.redirectTo,
       scopes: credentials.options?.scopes,
       queryParams: credentials.options?.queryParams,
       skipBrowserRedirect: credentials.options?.skipBrowserRedirect,
+      isPKCE: credentials.options?.isPKCE,
```

Comment:
Given that PKCE is an implementation detail, and probably the default flow, I'd maybe change the API here to be something like:

```
flowType: 'pkce' | 'implicit'
```

So that people can choose which flow they want, while `pkce` becomes the default flow.